### PR TITLE
FOUNDATIONS: Retrieve Group By Id

### PR DESCRIPTION
**Pull request by evangelinedrink**

### CODE REVIEW

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+using Taarafo.Portal.Web.Models.Groups.Exceptions;
+using Xunit;
+
+namespace Taarafo.Portal.Web.Tests.Unit.Services.Foundations.Groups
+{
+    public partial class GroupServiceTests
+    {
+        [Theory]
+        [MemberData(nameof(CriticalDependencyExceptions))]
+
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs:21]: remove extra line

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            this.apiBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowDependencyExceptionOnRetrieveByIdIfDependencyApiErrorOccursAndLogItAsync()
+        {
+            //given
+            Guid someGroupId = Guid.NewGuid();
+            var randomExceptionMessage = GetRandomMessage();
```
Comment (ShriHumrudha) [inline Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs:65]: string

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs (hunk 1 above)
Comment (evangelinedrink) [reply Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs]: I removed the extra line in the Code Rub.

File: Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs (hunk 2 above)
Comment (evangelinedrink) [reply Taarafo.Portal.Web.Tests.Unit/Services/Foundations/Groups/GroupServiceTests.Exceptions.RetrieveById.cs]: I changed the data type to String in the Code Rub.

**Review by ShriHumrudha — APPROVED** [review]

**Pull request merged**